include/deal.II/matrix_free/matrix_free.h

```diff
@@ -4525,8 +4596,17 @@ namespace internal
     cell(const std::pair<unsigned int, unsigned int> &cell_range) override
     {
       if (cell_function != nullptr && cell_range.second > cell_range.first)
-        (container.*
-         cell_function)(matrix_free, this->dst, this->src, cell_range);
+        for (unsigned int i = 0; i < matrix_free.n_active_fe_indices(); ++i)
+          {
+            const auto cell_subrange =
+              matrix_free.create_cell_subrange_hp_by_index(cell_range, i);
+
+            if (cell_subrange.second <= cell_subrange.first)
+              continue;
+
+            (container.*
+             cell_function)(matrix_free, this->dst, this->src, cell_subrange);
+          }
```

Comment:
Here and below the new/old/deprecated functions `MatrixFree::create_cell_subrange_hp_by_index()`, `MatrixFree::create_inner_face_subrange_hp_by_index()`,  `MatrixFree::create_boundary_face_subrange_hp_by_index()` are called. This is definitely not optimal. However, shows that we could introduce an additional loop level here obviously in smarter way (by pre-computed ranges).

Reply:
That's something we can do in a follow-up work.